vic/drivers/classic/src/write_data.c

```diff
@@ -59,10 +58,10 @@ write_data(stream_struct *stream,
         tmp_dptr = calloc(n, sizeof(*tmp_dptr));
 
         // Time
-        tmp_iptr[0] = dmy->year;
-        tmp_iptr[1] = dmy->month;
-        tmp_iptr[2] = dmy->day;
-        tmp_iptr[3] = dmy->dayseconds;
+        tmp_iptr[0] = stream->bounds[0].year;
+        tmp_iptr[1] = stream->bounds[0].month;
+        tmp_iptr[2] = stream->bounds[0].day;
+        tmp_iptr[3] = stream->bounds[0].dayseconds;
 
         // Write the date
         if (stream->agg_alarm.is_subdaily) {
```

Comment:
since everything is now consistently period-beginning, I am not convinced we need to treat daily output differently (basically `1990-01-01 00:00:00` is the same as `1990-01-01`. But leave as is for now